Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -114,41 +117,136 @@ namespace PhysX
     {
         const float expectedGravityMultiplier = 2.5f;
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
 
-        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight, DefaultFloorTransform);
+        // Set the gravity via the accessor
         basis.m_gameplayController->SetGravityMultiplier(expectedGravityMultiplier);
-        EXPECT_THAT(expectedGravityMultiplier, testing::FloatEq(basis.m_gameplayController->GetGravityMultiplier()));
+        EXPECT_THAT(basis.m_gameplayController->GetGravityMultiplier(), testing::FloatEq(expectedGravityMultiplier));
     }
 
+    //! Test that verifies that the Gravity Set & Get methods operate as expected while the scene ticks
+    //! and interacts with the entity & its components.
     TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_GravitySetsWhileMoving)
     {
         const float expectedGravityMultiplier = 2.5f;
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
 
-        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight, DefaultFloorTransform);
-
-        // Let scene run for a few moments so the entity can be manipulated by gravity from the gameplay component
+        // Let scene run for a few moments so the entity can be manipulated by gravity from the gameplay component.
         int duration = 10;
-
         for (int i = 0; i < duration; i++)
         {
             basis.m_gameplayController->SetGravityMultiplier(expectedGravityMultiplier + i);
             basis.Update();
 
-            EXPECT_THAT(expectedGravityMultiplier + i, testing::FloatEq(basis.m_gameplayController->GetGravityMultiplier()));
+            EXPECT_THAT(basis.m_gameplayController->GetGravityMultiplier(), testing::FloatEq(expectedGravityMultiplier + i));
+        }
+    }
+
+    //! Test that verifies that the Falling Velocity Set & Get methods function as expected.
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_FallingVelocitySets)
+    {
+        const AZ::Vector3 expectedVelocity(0.0f, 0.0f, 22.0f);
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        // Get the original velocity so we can determine that set actually changes things.
+        auto originalVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        // Set the falling velocity to the expected + original velocity to verify that the velociy will change.
+        basis.m_gameplayController->SetFallingVelocity(originalVelocity + expectedVelocity);
+
+        // Get the end velocity and validate that it is the expected value.
+        auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+        EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(expectedVelocity.GetZ(), 0.001f));
```

Comment:
The falling velocity should be whatever you set it to (immediately after you set it - it will change when you update physics). I think this test can only be passing because the original falling velocity was 0 or very close to it. If you've never ticked physics, then the original velocity would be 0.

I think a better test would be setting the falling velocity, ticking physics once and verifying that the new position was consistent with the velocity value you set.

Reply:
Reconfigured the logic here.

I was intending to use the starting velocity to shortcut setting the velocity to remove the starting velocity's default value as a failsafe for a false positive.

I've been more explicit about the testing of values at the end of the test.

As this test is trying to isolate whether or not the get & sets work outside of the physics tick modifying the velocity, I have not updated the physics tick and have another test that does this.

Reply:
OK, that makes sense with regard to not ticking the physics.

If you're worried about a false positive, in this case the original velocity should be 0, because physics has never ticked, so you could also just check that the original velocity was 0. The way you were doing things before just complicates the test, and before you explained why you were doing it, as someone reading the code I had no idea what the reason was for that complication. In general, I would say it's always better to aim for more explicitness and simplicity.

I would also argue that the scenario you're trying to guard against (that SetFallingVelocity is broken and GetVelocity just happens to return 22 by accident) is so unlikely that it's not worth testing for.

Reply:
removed that check & unused originalVelocity variable.